integration-tests/src/test/resources/apps/cdtservlet/src/application/CdtTxServlet.java

```diff
@@ -0,0 +1,249 @@
+    System.out.println("in cdttxservlet doGet method - domain2NsParam = " + domain2NsParam);
+
+    String domain2Url = "t3://domain2-managed-server1." + domain2NsParam + ","
+        + "domain2-managed-server2." + domain2NsParam + ":8001";
+    String tableName = "cdt_table";
+    java.sql.Connection conn;
+    Destination d;
+    Context ctx;
+    JMSContext context;
+
+    try {
+      DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+      java.util.Date date = new java.util.Date();
+      System.out.println("Today Time [" + dateFormat.format(date) + "]");
+      out.println("Today Time [" + dateFormat.format(date) + "]");
+      out.println("");
+
+      Hashtable h1 = new Hashtable();
+      h1.put(Context.INITIAL_CONTEXT_FACTORY,
+          "weblogic.jndi.WLInitialContextFactory");
+      String providerUrl = "t3://domain1-managed-server1." + domain1NsParam + ","
+          + "domain1-managed-server2." + domain1NsParam + ":8001";
+      System.out.println("BR: providerUrl = " + providerUrl);
+      h1.put(Context.PROVIDER_URL, providerUrl);
+
+      //Get the context from domain1. This is where JMS Queue is configured
+      ctx = new InitialContext(h1);
+      System.out.println("Got Local InitialContext [" + ctx + "]");
+      out.println("Got Local InitialContext [" + ctx + "]");
+
+      ConnectionFactory qcf = (ConnectionFactory) ctx.lookup("weblogic.jms.XAConnectionFactory");
+      d = (Destination) ctx.lookup("jms/testCdtUniformQueue");
+      System.out.println("Got Local JMS Destination " + d);
+      out.println("Got Local JMS Destination " + d);
+      context = qcf.createContext();
+
+      //Get context from domain2. This is where JDBC DS is configured
+      out.println("Getting Remote Context from " + domain2Url);
+      Hashtable h = new Hashtable();
+      h.put(Context.INITIAL_CONTEXT_FACTORY,
+          "weblogic.jndi.WLInitialContextFactory");
+      h.put(Context.PROVIDER_URL, domain2Url);
+
+      Context ctx2 = new InitialContext(h);
+      System.out.println("Got Remote InitialContext [" + ctx2 + "]");
+      out.println("Got Remote InitialContext [" + ctx2 + "]");
+
+      javax.sql.DataSource ds = (javax.sql.DataSource) ctx2.lookup("jdbc/TestCdtDataSource");
+      System.out.println("Got ds from context - " + ds);
+      out.println("Got ds from context - " + ds);
+      conn = ds.getConnection();
+      System.out.println("Got connection to datasource - " + conn);
+      out.println("Got connection to datasource - " + conn);
+
+      //create a table in the DB
+      createTable(conn, tableName);
+
+      try {
+        tm.begin();
+        Transaction tx = (Transaction) tm.getTransaction();
+        Xid xid = tx.getXID();
+        System.out.println("xid=" + xid);
+        tx.setProperty("TMAfterTLogBeforeCommitExit", Boolean.TRUE);
+
+        ds = (javax.sql.DataSource) ctx2.lookup("jdbc/TestCdtDataSource");
```

Comment:
You don't need to lookup the datasource again here.  Getting the connection under a global tx will cause the JDBC resource to be enlisted in the tx.

Reply:
Fixed